crates/bevy_gizmos/src/gizmos.rs

```diff
@@ -829,7 +829,7 @@ where
             self.gizmos
                 .circle(
                     self.position,
-                    Dir3::new_unchecked(self.rotation * axis),
+                    Dir3::new(self.rotation * axis).unwrap(),
```

Comment:
Not a fan of calling `.unwrap()` here since we want to minimise engine-panics, could we handle the error (when the value is approx. 0) differently? Maybe `.unwrap_or(some_default_value)`.

Reply:
I think it's better to handle approximations of zero internally within `Dir3::new`, and for other cases, simply take `Dir3::Z` as the default value?

Reply:
I'm not against trying to handle this error with something like `unwrap_or`, but I also think it will literally never happen; I'd be surprised if it couldn't be proven to be impossible given that the rotation and axis are both already normalized.

Reply:
> I'm not against trying to handle this error with something like `unwrap_or`, but I also think it will literally never happen; I'd be surprised if it couldn't be proven to be impossible given that the rotation and axis are both already normalized.

Is the rotation normalized here, and if it is, it doesn't seem like there is a need for this PR, but I encountered a panic here in version 0.13.2, and the judgment normalized was a bit stricter at the time

Reply:
> My understanding of quaternions was insufficient; if the quaternion is normalized, it seems that this PR would not be necessary.

The issue (as I understand it) is that even if `self.rotation` and `axis` are both normalized, the result of `self.rotation * axis` is only necessarily almost-normalized because of floating point error.

Reply:
> The issue (as I understand it) is that even if `self.rotation` and `axis` are both normalized, the result of `self.rotation * axis` is only necessarily almost-normalized (because of floating point error).

Indeed, that seems to be the case. So this PR stems from my misunderstanding, and I will close it.